### PR TITLE
Feat: Feedback/Refactor

### PR DESCRIPTION
**Pull request by damiramirez**
**This PR**

- Put code blocks in README.
- Modify the .PHONY in the Makefile.
- Change the indexing in bounds check to use `.get()`.
- Change the logic of how the loop game renders the board.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -82,10 +84,11 @@ async fn main() {
             conways.toggle_state_cell((mark_cell_x, mark_cell_y));
         }
 
-        if running && get_time() - last_updated > UPDATE_TIMER {
-            last_updated = get_time();
+        if running && accum_delta >= 2. * TICK_RATE {
```
Comment (LeanSerra) [inline src/main.rs]: Why is this checking if the delta is twice the tick_rate, wouldn't it be easier double the `TICK_RATE` const?

**Review by LeanSerra — CHANGES_REQUESTED** [review]

**Comment by lima-limon-inc** [comment]
On it

File: src/main.rs (hunk 1 above)
Comment (damiramirez) [reply src/main.rs]: My mistake, I forgot to remove that hehe

**Review by LeanSerra — APPROVED** [review]
LGTM

File: src/conways.rs [hunk 2]
```diff
@@ -85,33 +83,30 @@ impl Conways {
...
-            if new_x >= 0
-                && new_x < ROWS as isize
-                && new_y >= 0
-                && new_y < COLUMNS as isize
-                && self.grid[new_x as usize][new_y as usize] == CellState::Alive
+            if self
+                .grid
+                .get(new_x as usize)
+                .and_then(|column| column.get(new_y as usize))
+                .is_some_and(|state| *state == CellState::Alive)
```
Comment (lima-limon-inc) [inline src/conways.rs:90]: This is lovely! Very nice

File: src/conways.rs [hunk 3]
```diff
@@ -15,19 +15,21 @@ pub struct Conways {
...
-impl Conways {
-    pub fn from(alive_cells: Vec<Position>) -> Self {
+impl From<Vec<Position>> for Conways {
+    fn from(alive_cells: Vec<Position>) -> Self {
         let mut grid = vec![vec![CellState::Dead; COLUMNS]; ROWS];
 
         for (x, y) in alive_cells {
-            if x < ROWS && y < COLUMNS {
-                grid[x][y] = CellState::Alive;
+            if let Some(cell) = grid.get_mut(x).and_then(|column| column.get_mut(y)) {
```
Comment (lima-limon-inc) [inline src/conways.rs:23]: Very nice!

**Review by lima-limon-inc — APPROVED** [review]
Very very nice! Lovely use of combinators

**Pull request merged**